sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -573,4 +573,48 @@ object DateTimeUtils {
       dayInYear - 334
     }
   }
+
+  /**
+   * Returns Day of week from String. Starting from friday, marked as 0.
+   */
+  def getDayOfWeekFromString(string: UTF8String): Int = {
+    val dowString = string.toString.toUpperCase
+    dowString match {
+      case "SU" | "SUN" | "SUNDAY" => 2
+      case "MO" | "MON" | "MONDAY" => 3
+      case "TU" | "TUE" | "TUESDAY" => 4
+      case "WE" | "WED" | "WEDNESDAY" => 5
+      case "TH" | "THU" | "THURSDAY" => 6
+      case "FR" | "FRI" | "FRIDAY" => 0
+      case "SA" | "SAT" | "SATURDAY" => 1
+      case _ => -1
+    }
+  }
+
+  /**
+   * Returns the first date which is later than startDate and is of the given dayOfWeek.
+   */
+  def getNextDateForDayOfWeek(startDate: Int, dayOfWeek: Int): Int = {
```

Comment:
we should document what dayOfWeek here means, i.e. in this case, Fri is 0, Sat is 1 ...